src/main/kotlin/com/exactpro/th2/common/value/ValueUtils.kt

```diff
@@ -30,34 +30,34 @@ import java.math.BigInteger
 fun nullValue(): Value = Value.newBuilder().setNullValue(NULL_VALUE).build()
 fun listValue() : ListValue.Builder = ListValue.newBuilder()
 
-fun Value.getString(): String? = if (this.kindCase == SIMPLE_VALUE) this.simpleValue else null
+fun Value.getString(): String? = takeIf { kindCase == SIMPLE_VALUE }?.simpleValue
 fun Value.getInt(): Int? = this.getString()?.toIntOrNull()
 fun Value.getLong(): Long? = this.getString()?.toLongOrNull()
 fun Value.getDouble(): Double? = this.getString()?.toDoubleOrNull()
 fun Value.getBigInteger(): BigInteger? = this.getString()?.toBigIntegerOrNull()
 fun Value.getBigDecimal(): BigDecimal? = this.getString()?.toBigDecimalOrNull()
-fun Value.getMessage(): Message? = if (hasMessageValue()) this.messageValue else null
-fun Value.getList() : List<Value>? = if (hasListValue()) this.listValue.valuesList else null
+fun Value.getMessage(): Message? = takeIf(Value::hasMessageValue)?.messageValue
+fun Value.getList() : List<Value>? = takeIf(Value::hasListValue)?.listValue?.valuesList
 
-fun Value.Builder.updateList(updateFunc: ListValue.Builder.() -> ListValueOrBuilder?) : Value.Builder = apply { if (hasListValue()) updateOrAddList(updateFunc) else throw NullPointerException("Can not find list value") }
-fun Value.Builder.updateValue(updateFunc: String.() -> String?) : ValueOrBuilder = apply { simpleValue = updateFunc(simpleValue ?: throw NullPointerException("Can not find simple value")) }
-fun Value.Builder.updateMessage(updateFunc: Message.Builder.() -> MessageOrBuilder?) : Value.Builder = apply { if(hasMessageValue()) updateOrAddMessage(updateFunc) else throw NullPointerException("Can not find message value") }
+fun Value.Builder.updateList(updateFunc: ListValue.Builder.() -> ListValueOrBuilder) : Value.Builder = apply { check(!hasListValue()) { "Can not find list value" }; updateOrAddList(updateFunc) }
+fun Value.Builder.updateValue(updateFunc: String.() -> String) : ValueOrBuilder = apply { simpleValue = updateFunc(simpleValue ?: throw NullPointerException("Can not find simple value")) }
+fun Value.Builder.updateMessage(updateFunc: Message.Builder.() -> MessageOrBuilder) : Value.Builder = apply { check(!hasMessageValue()) { "Can not find message value" }; updateOrAddMessage(updateFunc) }
 
-fun Value.Builder.updateOrAddList(updateFunc: ListValue.Builder.() -> ListValueOrBuilder?) : Value.Builder = apply { updateFunc(listValueBuilder)?.also {
+fun Value.Builder.updateOrAddList(updateFunc: ListValue.Builder.() -> ListValueOrBuilder) : Value.Builder = apply { updateFunc(listValueBuilder).also {
     when (it) {
         is ListValue -> listValue = it
         is ListValue.Builder -> setListValue(it)
-        else -> throw IllegalStateException("Can not set list value")
+        else -> throw IllegalStateException("Can not set list value. Wrong type = ${it.javaClass.typeName}")
```

Comment:
```suggestion
        else -> error("Can not set list value. Wrong type = ${it.javaClass.typeName}")
```

Reply:
Also why `typeName` instead of `canonicalName` / `qualifiedName`? And it's better to use `::class.java` instead of `.javaClass` because it could lead to an unexpected result inside of companion-object (which is not the case here, but still)